src/librustc_codegen_llvm/back/write.rs

```diff
@@ -440,11 +438,10 @@ pub(crate) unsafe fn optimize(cgcx: &CodegenContext<LlvmCodegenBackend>,
 pub(crate) unsafe fn codegen(cgcx: &CodegenContext<LlvmCodegenBackend>,
                   diag_handler: &Handler,
                   module: ModuleCodegen<ModuleLlvm>,
-                  config: &ModuleConfig,
-                  timeline: &mut Timeline)
+                  config: &ModuleConfig)
     -> Result<CompiledModule, FatalError>
 {
-    timeline.record("codegen");
+    cgcx.profile(|p| p.generic_event("codegen"));
```

Comment:
`start-end` activity that spans the entire function body of `fn codegen()`. The sub events in here can also be converted into `start-end` events or removed for now.